src/lib/utils/fetchTokenList.ts

```diff
@@ -0,0 +1,128 @@
+import type { TokenInfo, TokenList } from '@uniswap/token-lists'
+import type { Ajv, ValidateFunction } from 'ajv'
+
+import contenthashToUri from './contenthashToUri'
+import parseENSAddress from './parseENSAddress'
+import uriToHttp from './uriToHttp'
+
+export const DEFAULT_TOKEN_LIST = 'https://gateway.ipfs.io/ipns/tokens.uniswap.org'
+
+enum ValidationSchema {
+  LIST = 'list',
+  INFO = 'info',
+}
+
+/**
+ * Returns a ValidateFunction for token lists.
+ * Lazily loads ajv and the schema for performance.
+ */
+const getValidateFunction = (() => {
```

Comment:
see https://github.com/Uniswap/interface/pull/3080/files#r780404165

Reply:
src/utils/fetchTokenList.ts has been renamed to this file, so no duplication here :)